crates/rmcp/src/service/server.rs

```diff
@@ -3,6 +3,34 @@ use std::borrow::Cow;
 use thiserror::Error;
 
 use super::*;
+
+/// Validates timeout values to prevent DoS attacks and ensure reasonable limits
+fn validate_timeout(timeout: Option<std::time::Duration>) -> Result<(), ServiceError> {
```

Comment:
Hmm is this necessary? Why does it need to be limited to 1ms-5min? Values outside of that might be very uncommon but I'd think that should be left to the server developer.